Library/Homebrew/cmd/bottle.rb

```diff
@@ -183,6 +186,15 @@ def bottle_formula(f)
     relocatable = false
     skip_relocation = false
 
+    formula_source_time = f.stable.stage do
+      newest_mtime = nil
+      Pathname(Dir.pwd).find do |path|
+        newest_mtime ||= path.mtime
+        newest_mtime = path.mtime if path.mtime > newest_mtime
+      end
+      newest_mtime
+    end
```

Comment:
> More importantly, this would mean network is required for bottle, which may be something we want to avoid. Especially for formulae who are huge.

If you've done `brew install --build-bottle` before this you'll have a cached download so shouldn't need the network.



> Finally, git doesn't store timestamp. So we will get different timestamp for different stage for formulae using GitDownloadStrategy. In such case, we should use git commit date as the timestamp. And we probably need to check other download strategies as well. Therefore, it may be more help to introduce AbstractDownloadStrategy#newest_mtime method. If so, putting logic in formula_installer make more sense.

Yup, that's another edge-case to be addressed (in a future PR if that's :ok:)